Support React 17

Currently, `react-virtualized` has a peer dependency on `react` and `react-dom` that constraints it to React 16. From my experiments, it already seems to work with React 17 so this might be as simple as just bumping the bounds.

This is getting more important when using React 17 and upgrading to npm 7.x, as there is a [breaking change](https://github.com/npm/cli/blob/latest/CHANGELOG.md#dependency-resolution) about the peer dependency resolution, resulting in
```
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^15.3.0 || ^16.0.0-alpha" from react-virtualized@9.22.3
npm ERR! node_modules/react-virtualized
npm ERR!   react-virtualized@"9.22.3" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

the workaround is using `--legacy-peer-deps`, but it would be great if react-virtualized would officially support React 17.

--- a/package.json
+++ b/package.json
@@ -118,12 +118,12 @@
     "prettier": "1.19.1",
     "pretty-quick": "^2.0.1",
     "puppeteer": "^2.0.0",
-    "react": "^16.11.0",
+    "react": "^17.0.1",
     "react-codemirror": "^1.0.0",
-    "react-dom": "^16.11.0",
-    "react-router": "^5.1.2",
-    "react-router-dom": "^5.1.2",
-    "react-test-renderer": "^16.11.0",
+    "react-dom": "^17.0.1",
+    "react-router": "^5.2.0",
+    "react-router-dom": "^5.2.0",
+    "react-test-renderer": "^17.0.1",
     "rimraf": "^3.0.0",
     "rollup": "^1.26.5",
     "rollup-plugin-babel": "^4.3.3",
@@ -146,8 +146,8 @@
     "react-lifecycles-compat": "^3.0.4"
   },
   "peerDependencies": {
-    "react": "^15.3.0 || ^16.0.0-alpha",
-    "react-dom": "^15.3.0 || ^16.0.0-alpha"
+    "react": "^15.3.0 || ^16.0.0-alpha || ^17.0.1",
+    "react-dom": "^15.3.0 || ^16.0.0-alpha || ^17.0.1"
   },
   "browserify": {
     "transform": [

